### ISSUE TITLE
Reverse proxy problems.

### ISSUE THREAD

**Issue by zZHorizonZz** [vercel/next.js#60956]
### Link to the code that reproduces this issue

https://github.com/zZHorizonZz/nextjs-routing

### To Reproduce

1. Start the application
2. Create a new reverse proxy nginx or IIS and have it route requests for path test for example. (In the repro repository there is an attached file for IIS page config with rewrite rule configuration)
3. Try to access the site 

### Current vs. Expected behavior

Current behavior works like this:
1) IIS Rewrite test -> localhost:3000 (http://localhost/test) This will get the index file but other files are not found as it's pointing to localhost/_next
2) IIS Rewrite test -> localhost:3000 with basePath set to test (http://localhost/test) no file is found even the index file (404)
3) IIS Rewrite test -> localhost:3000/test with basePath set to test (http://localhost/test) results in error too many redirects

The expected behavior is this should probably route and return the correct files. I found a related discussion to this [here](https://github.com/vercel/next.js/discussions/59904)

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Enterprise
Binaries:
  Node: 20.9.0
  npm: N/A
  Yarn: N/A
  pnpm: 8.6.2
Relevant Packages:
  next: 14.1.0
  eslint-config-next: 14.1.0
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

App Router, Routing (next/router, next/navigation, next/link)

### Which stage(s) are affected? (Select all that apply)

next dev (local), next build (local), next start (local)

**Comment by gitgetgud** [comment]
Facing similar issues with nginx and nextjs 14.1.0

**Comment by gitgetgud** [comment]
[Duplicate?](https://github.com/vercel/next.js/issues/41824) Removing trailing / fixed my issue

**Comment by zZHorizonZz** [comment]
> [Duplicate?](https://github.com/vercel/next.js/issues/41824) Removing trailing / fixed my issue

I don't think this is related to my case as I don't have any trailing / in any of the settings.

**Comment by zZHorizonZz** [comment]
After looking into it more closely, I've noticed something unusual. Even the default application returns a 404 error, but the issue is more complex. We receive a valid index response, which includes references to subsequent scripts and other elements, so the responses themselves appear valid. However, something occurs subsequently, and Next.js sets the status of response to 404. The only explanation I can think of is that there might be an issue with invalid headers. Could this be the reason Next.js is marking the request as 404? But that's just a wild guess.

**Comment by willmeierart** [comment]
> After looking into it more closely, I've noticed something unusual. Even the default application returns a 404 error, but the issue is more complex. We receive a valid index response, which includes references to subsequent scripts and other elements, so the responses themselves appear valid. However, something occurs subsequently, and Next.js sets the status of response to 404. The only explanation I can think of is that there might be an issue with invalid headers. Could this be the reason Next.js is marking the request as 404? But that's just a wild guess.

do the scripts in the index all have the appropriate basePath prepended?

**Comment by zZHorizonZz** [comment]
> > After looking into it more closely, I've noticed something unusual. Even the default application returns a 404 error, but the issue is more complex. We receive a valid index response, which includes references to subsequent scripts and other elements, so the responses themselves appear valid. However, something occurs subsequently, and Next.js sets the status of response to 404. The only explanation I can think of is that there might be an issue with invalid headers. Could this be the reason Next.js is marking the request as 404? But that's just a wild guess.
> 
> do the scripts in the index all have the appropriate basePath prepended?

Yes, it does, but I have noticed that when I go to the default path, Next.js starts building the not-found page, which seems weird.

**Comment by encloinc** [comment]
I also have this issue..

**Comment by uchar** [comment]
In my case this one first redirect correctly to next.js but then nextjs somehow rermove /panel and send request again
```
location ~^/panel(.*)$ {
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header Host ....com;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    set $upstream http://user_panel:3000;
    proxy_pass $upstream$1$is_args$args;
}
```
but  this one works just fine :
```
location /panel {
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header Host ....com;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_pass http://user_panel:3000;
}
```

**Comment by hypo-thesis** [comment]
the way I did through Apache is 

```
#PROXY-START/
<IfModule mod_proxy.c>

  ProxyRequests Off
    ProxyPreserveHost On
    AllowEncodedSlashes NoDecode 

    ProxyPass / http://127.0.0.1:3210/ nocanon
    ProxyPassReverse / http://127.0.0.1:3210/

    </IfModule>
#PROXY-END/
```

Thought might come in handy.

**Comment by marcoshernanz** [comment]
Thanks for the report. I reproduced the three cases, and they come from how the proxy rewrites the request path. With `basePath: '/test'`, Next.js must receive `/test`. Stripping it sends `/` and returns 404, while proxying to `/test/` creates a trailing-slash redirect loop. Preserving the path fixes the page and assets on 14.1.0 and current canary. The original repro is no longer available, so I’m closing this. If this still fails with the prefix preserved, please open a new issue with a working repro and full proxy config.
